### ISSUE TITLE
[Question] Close all available {browser} contexts

### ISSUE THREAD

**Issue by gw-alexniculae** [microsoft/playwright#27796]
Hi,

Could you please share if you have a solution to close all browser contexts in one shot? (multiple contexts created for a test with multiple logins)

I tried looking into browser.contexts(), but this cannot be used as it's not a promise.

I'm using playwright test, latest version.

Thank you!

**Comment by yury-s** [comment]
> I tried looking into browser.contexts(), but this cannot be used as it's not a promise.

Can you elaborate on why iterating all the contexts does not work?

Another option would be to just call `browser.close()`, not sure if it works for you. Also you can create contexts as [fixtures](https://playwright.dev/docs/test-fixtures#creating-a-fixture) which could automatically close them.

**Comment by gw-alexniculae** [comment]
Thanks for the quick answer @yury-s

I did try to iterate through them, but the tests will fail as the browser seems to close before the test finishes.

I also don't really understand why that happens, but running a forEach on each context will error the tests.

I'll try to work on a reproducible example, but in the meanwhile, I decided to try the fixtures.

I read on PW docs that browser.close() won't handle the contexts gracefully, so I didn't try that.
